cli/tools/lint/mod.rs

```diff
@@ -293,8 +293,10 @@ impl WorkspaceLinter {
       deno_lint_config: lint_config,
     }));
 
+    eprintln!("paths {:#?}", paths);
     let mut futures = Vec::with_capacity(2);
-    if linter.has_package_rules() {
+    // Package rules are only in effect if user didn't explicitly specify paths on the command line
+    if paths.is_empty() && linter.has_package_rules() {
```

Comment:
This bit seems broken - whether we invoke `deno lint` or `deno lint main.ts` in a directory that has `deno.json`, the `paths` contains `main.ts`. We were unconditionally running the "package rules" lints as long as the `deno.json` contains "publish config". Changing this conditional effectively breaks package lints. I think this whole bit should be rewritten to account for the fact that we're running lint for a whole dir or a certain file.